fakebook/src/app/component/newsfeed/newsfeed.component.ts

```diff
@@ -13,43 +13,42 @@ import { PostService } from '../../services/post.service';
 export class NewsfeedComponent implements OnInit {
   posts: Post[] = [];
   user: User | null = null;
-  postId: number;
+  postId: number | undefined;
 
   constructor(
-    private newsfeedService: NewsfeedService,
-    private postService: PostService,
-    private route: ActivatedRoute
-  ) {
-    this.route.params.subscribe(params => {
-      console.log(params);
-      if(params['id'] != undefined){   
+    private readonly newsfeedService: NewsfeedService,
+    private readonly postService: PostService,
+    private readonly route: ActivatedRoute
+    ) {
+    this.route.params.subscribe((params) => {
+      if(params['id'] !== undefined) {
         this.postId = +params['id'];
-          if(this.postId != undefined)
-            this.getPostsById();
-      }
-      else{
+        if(this.postId !== undefined && !isNaN(this.postId)) {
+          this.getPostById();
+        }
+      } else {
         this.getPosts();
       }
-    })
+    });
   }
 
   ngOnInit(): void {
     this.getUser();
-    //this.getPosts();
```

Comment:
do we not want to get posts on init? how else will the newsfeed show things? i might be missing something

Reply:
the get post logic was in the constructor. I've changed it to be in the on init.